### PR TITLE
As an API user, I can log in and get the tokens

### LINKED ISSUE #14: As an API user, I can log in and get the tokens
### Description
- [ ] Create `POST /api/v1/oauth/token`
- [ ] Use OAuth2
- [ ] Support refresh_token

### Acceptance Criteria
- Able to get the access_token

### PR DESCRIPTION
**Pull request by hoangmirs**
Resolve #14

## What happened 👀

Add `/token` endpoint to get and refresh token

## Insight 📝

- Use the built-in function of OAuth2 lib
- Support 2 `grant_type`: `password`, `refresh_token`
- Submit with form data:
```
client_id:{{client_id}}
client_secret:{{client_secret}}
grant_type:password/refresh_token
username:{{email}}
password:{{password}}
```

## Proof Of Work 📹

Update Postman collection later

_Request access token:_
Postman

_Refresh token:_
Postman

### CODE REVIEW

**Review by Lahphim — APPROVED** [review]

**Review by olivierobert — APPROVED** [review]

File: controllers/api/v1/oauth_token.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+)
+
+type OAuthToken struct {
+	baseController
+}
+
+func (c *OAuthToken) Post() {
+	server := oauth.GetOAuthServer()
+
+	server.SetPasswordAuthorizationHandler(passwordAuthorizationHandler)
```
Comment (junan) [inline controllers/api/v1/oauth_token.go:19]: This setting handler actually needs to run only once. What about to place it in the `init` function or inilializer/boostrap?

File: controllers/api/v1/oauth_token.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+	err := server.HandleTokenRequest(c.Ctx.ResponseWriter, c.Ctx.Request)
```
Comment (junan) [inline controllers/api/v1/oauth_token.go:21]: With this it renders the response which is non JSON API structure, to fix this issue I had to create a [new writer](https://github.com/junan/google-scraper/blob/23dafc318655e5fc939e0b33d9686c13e191264c/controllers/api/v1/token.go#L19)

Comment (hoangmirs) [reply controllers/api/v1/oauth_token.go:16]: Thank you, let me check and implement it in another PR.
I created https://github.com/hoangmirs/go-scraper/issues/59 to do that

File: controllers/api/v1/oauth_token_test.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+						"client_secret": {"invalid"},
+						"grant_type":    {"password"},
+						"username":      {email},
+						"password":      {password},
+					}
+					body := strings.NewReader(form.Encode())
+
+					response := MakeRequest("POST", "/api/v1/oauth/token", body)
+
+					Expect(response.Code).To(Equal(http.StatusInternalServerError))
```
[the remark this answers is not in the document]
Comment (hoangmirs) [reply controllers/api/v1/oauth_token_test.go:118]: This error was raised by `oServer.HandleTokenRequest`, I will keep it for now and if possible, I will update it when customizing the OAuth response

**Review by junan — APPROVED** [review]

**Pull request merged**